functional-test/src/test/groovy/org/rundeck/util/common/jobs/JobUtils.groovy

```diff
@@ -155,35 +152,33 @@ class JobUtils {
      *
      * @param expectedStates A list of expected states.
      * @param executionId The execution ID to query.
-     * @param mapper The ObjectMapper instance to parse the response.
      * @param client The RdClient instance to perform the HTTP request.
-     * @param iterationGap The duration to wait between each check.
      * @param timeout The maximum duration to wait for the execution to reach the expected state.
+     * @param checkPeriod The time to wait between each check.
      * @return The Execution object representing the execution status.
      * @throws InterruptedException if the timeout is reached before the execution reaches the expected state.
      */
-    static Execution waitForExecutionToBe(
+    static Execution waitForExecution(
         Collection<String> expectedStates,
         String executionId,
-        ObjectMapper mapper,
         RdClient client,
-        Duration iterationGap = WaitingTime.LOW,
-        Duration timeout = WaitingTime.MODERATE
+        Duration timeout = WaitingTime.MODERATE,
+        Duration checkPeriod = WaitingTime.LOW
     ) {
         def execDetail = client.doGet("/execution/${executionId}")
-        Execution executionStatus = mapper.readValue(execDetail.body().string(), Execution.class)
+        Execution executionStatus = OBJECT_MAPPER.readValue(execDetail.body().string(), Execution.class)
         long initTime = System.currentTimeMillis()
```

Comment:
in 2010, it was strongly discouraged to implement your own thread pool.
in 2020s, it should be discouraged to to implement your own retry logic.
I'm sure there multiple java libs that will turn this while loop into something similar to:
```
Retry.timeout(15).retryEvery(5) { code_to_retry() }
```

Reply:
I agree with this, but for the moment i don't want to add a new dependency. I couldn't find an existing api that would leverage this within our stack, but probably we could explore spring retry: https://www.baeldung.com/spring-retry

Reply:
Since the scope of this PR has grown too big, i'll be implementing these improvements in a different PR